diff/testdata/sample_hunk_lines_start_with_minuses.diff

```diff
@@ -0,0 +1,5 @@
+@@ -1,4 +1,3 @@
+ select 1;
+--- this is my query
+ select 2;
+ select 3;
```

Comment:
```suggestion
@@ -1,4 +1,3 @@
 select 1;
--- this is my query
+++ this could break?
 select 2;
 select 3;
```

Reply:
Is that a case that works?

Reply:
You mean, with the fix here or in general?

Reply:
With the logic above that checks the next hunk line for `+++` and fails if it finds it.
Like when I change a line prefix from `--` to `++`, the diff would look like

```diff
line
---line
+++line
line
```
I guess. Would this be parseable?

Reply:
If I use this diff, it crashes:

```diff
diff --git a/foobar.sql b/foobar.sql
index 9537d7b..f73c856 100644
--- a/foobar.sql
+++ b/foobar.sql
@@ -1,5 +1,5 @@
 select 1;
--- this is my query
+++ this is my query
 select 2;
 select 3;
--- this is the last line
+++ this is the last line
```

I'll try to see if I can make that work.

Reply:
@keegancsmith @eseliger what do you think of this? https://github.com/sourcegraph/go-diff/pull/56

It adds a custom type, `lineReader`, so we can do two-line lookahead and check whether the `---` and `+++` are followed by a `@@ `, which is what `git` does. It's built on top of this PR and all tests pass, including the case raised here by @eseliger.